### PR TITLE
fix: display specific error message for wrong chain and non-owner wallets

### LINKED ISSUE #769: [Tx modal] Show specific error when trying to submit a tx while on a different chain
## Bug description
When trying to submit a tx on a safe while connected to a different chain no accurate description is shown to the user as in legacy.

## Steps to reproduce
Try to perform a transaction on a mainnet safe while being connected to Rinkeby on Metamask (for example).

## Expected result
(legacy)
Screenshot 2022-09-29 at 17 42 12

## Obtained result
(revamp)
Screenshot 2022-09-29 at 17 43 57

### PR DESCRIPTION
**Pull request by niklasnatter**
## What it solves

This PR displays a specific error message in the transaction modal if the connected wallet is connected to the wrong chain or if it is not an owner of the wallet. 

Resolves #769, #722

## How this PR fixes it

If the connected wallet is connected to the wrong chain or is not an owner, an explicit error message that explains the problem is displayed above the generic `This transaction will most likely fail.` message. Additionally, it disables the submit button in these cases. I used the error messages from #769 and #722.

## How to test it

1. Create a transaction in the transaction builder
2. Switch the connected wallet to a different chain or connect an unrelated address
3. Open the transaction overlay via the `Send Batch` button

## Screenshots
Screenshot 2022-10-07 at 16 08 36
Screenshot 2022-10-07 at 16 09 14

### CODE REVIEW

File: src/components/tx/SignOrExecuteForm/index.tsx [hunk 1]
```diff
@@ -164,7 +164,9 @@ const SignOrExecuteForm = ({
     setAdvancedParams(data)
   }
 
-  const submitDisabled = !isSubmittable || isEstimating || !tx
+  const walletIsWrongChain = currentChain?.chainId != wallet?.chainId
```
Comment (katspaugh) [inline src/components/tx/SignOrExecuteForm/index.tsx:167]: We've just merged another PR that adds `const isWrongChain = useIsWrongChain()`. Please pull from dev, you can then use this `isWrongChain` var.

File: src/components/tx/SignOrExecuteForm/index.tsx [hunk 2]
```diff
@@ -164,7 +164,9 @@ const SignOrExecuteForm = ({
...
+  const walletIsNoOwner = !safe.owners.some((owner) => owner.value === wallet?.address)
```
Comment (katspaugh) [inline src/components/tx/SignOrExecuteForm/index.tsx:168]: Non-owners can execute transactions, thought. So I would suggest to do smth like 
```
  const walletCanSubmit = willExecute || isSignableBy(transaction, wallet?.address || '')
```

File: src/components/tx/SignOrExecuteForm/index.tsx [hunk 3]
```diff
@@ -191,6 +193,14 @@ const SignOrExecuteForm = ({
           disabled={submitDisabled}
         />
 
+        {walletIsWrongChain && <ErrorMessage>Your wallet is connected to the wrong chain.</ErrorMessage>}
+
+        {walletIsNoOwner && (
+          <ErrorMessage>
+            You are currently not an owner of this Safe and won&apos;t be able to submit this transaction.
+          </ErrorMessage>
+        )}
```
Comment (katspaugh) [inline src/components/tx/SignOrExecuteForm/index.tsx:202]: That's nice 👍

**Review by katspaugh — COMMENTED** [review]
Many thanks Niklas! 🙏

File: src/components/tx/SignOrExecuteForm/index.tsx (hunk 2 above)
Comment (katspaugh) [reply src/components/tx/SignOrExecuteForm/index.tsx:168]: However, if it's a new tx (no `txId` passed), your check will work fine.
So to break down:

* New tx – must be an owner
* Existing tx to be signed – needs to be an owner who hasn't signed yet
* Existing tx to be executed – can be anyone

Comment (niklasnatter) [reply src/components/tx/SignOrExecuteForm/index.tsx]: Thanks for the feedback! I added the suggestions and fixed the testcases.

I am not familiar enough with the domain logic here- does your check already cover all 3 requirements or do i need to check if `txId` is passed additionally?

Comment (katspaugh) [reply src/components/tx/SignOrExecuteForm/index.tsx]: I think your current version should work as it is 👍

File: src/components/tx/SignOrExecuteForm/index.tsx [hunk 4]
```diff
@@ -164,7 +165,9 @@ const SignOrExecuteForm = ({
     setAdvancedParams(data)
   }
 
-  const submitDisabled = !isSubmittable || isEstimating || !tx
+  const isWrongChain = useIsWrongChain()
```
Comment (katspaugh) [inline src/components/tx/SignOrExecuteForm/index.tsx:168]: I've just realized this branch is based off of `main` but we've recently switched to `dev` as our default branch.
Could you please rebase to dev and also change the base branch of the PR? Then this `useIsWrongChain` will be there, it's on dev.

**Review by katspaugh — APPROVED** [review]
LGTM, thank you Niklas!

**Pull request merged**